app/models/spree/stock/estimator_decorator.rb

```diff
@@ -1,68 +1,58 @@
-    parcel = build_parcel(package)
-    shipment = build_shipment(from_address, to_address, parcel)
-    rates = shipment.rates.sort_by { |r| r.rate.to_i }
-
-    if rates.any?
-      rates.each do |rate|
-        package.shipping_rates << Spree::ShippingRate.new(
-          name: "#{ rate.carrier } #{ rate.service }",
-          cost: rate.rate,
-          easy_post_shipment_id: rate.shipment_id,
-          easy_post_rate_id: rate.id,
-          shipping_method: find_or_create_shipping_method(rate)
-        )
+module Spree
+  module Stock
+    module EstimatorDecorator
+      def shipping_rates(package)
+        shipment = package.easypost_shipment
+        rates = shipment.rates.sort_by { |r| r.rate.to_i }
+
+        if rates.any?
+          rates.each do |rate|
+            spree_rate = Spree::ShippingRate.new(
+              name: "#{ rate.carrier } #{ rate.service }",
+              cost: rate.rate,
+              easy_post_shipment_id: rate.shipment_id,
+              easy_post_rate_id: rate.id,
+              shipping_method: find_or_create_shipping_method(rate)
+            )
+
+            package.shipping_rates << spree_rate if spree_rate.shipping_method.frontend?
+          end
+
+          # Sets cheapest rate to be selected by default
+          if package.shipping_rates.any?
+            package.shipping_rates.min_by(&:cost).selected = true
+          end
+
+          package.shipping_rates
+        else
+          []
+        end
       end
 
-      # Sets cheapest rate to be selected by default
-      package.shipping_rates.first.selected = true
-
-      package.shipping_rates
-    else
-      []
-    end
-  end
-
-  private
-
-  # Cartons require shipping methods to be present, This will lookup a
-  # Shipping method based on the admin(internal)_name. This is not user facing
-  # and should not be changed in the admin.
-  def find_or_create_shipping_method(rate)
-    method_name = "#{ rate.carrier } #{ rate.service }"
-
-    unless shipping_method = Spree::ShippingMethod.find_by(admin_name: method_name)
-      Spree::ShippingMethod.create(
-        name: method_name,
-        admin_name: method_name,
-        display_on: :back_end,
-        code: rate.service,
-        calculator: Spree::Calculator::Shipping::FlatRate.create,
-        shipping_categories: [Spree::ShippingCategory.first]
-      )
-    end
-  end
-
-  def build_parcel(package)
-    total_weight = package.contents.sum do |item|
-      item.quantity * item.variant.weight
+      private
+
+      # Cartons require shipping methods to be present, This will lookup a
+      # Shipping method based on the admin(internal)_name. This is not user facing
+      # and should not be changed in the admin.
+      def find_or_create_shipping_method(rate)
+        method_name = "#{ rate.carrier } #{ rate.service }"
+        shipping_method = Spree::ShippingMethod.find_or_initialize_by(admin_name: method_name)
+
+        unless shipping_method.persisted?
```

Comment:
You can skip the local variable and persisted? check by doing find_or_create_by(admin_name: method_name) and passing in a block for the other attributes.